tools/python/boutiques/localExec.py

```diff
@@ -1155,6 +1177,112 @@ def check(keyname, isGood, msg, value):  # Checks input values
                 message += ("\t" + err + "\n")
             raise_error(ExecutorError, message)
 
+    # Private method to generate summary object of data
+    # collection file containing the tool name and descriptor DOI
+    def _generateSummary(self, desc):
+        summary = {}
+        summary['name'] = self.desc_dict['name']
+        summary['descriptor-doi'] = self._findDOI(desc)
+        return summary
+
+    # Private method to attempt to find descriptor DOI
+    # through various cases
+    # userInput may be json string, filename or zenodo id
+    def _findDOI(self, userInput):
+        # DOI in Zenodo reference case
+        if userInput.split(".")[0].lower() == "zenodo":
+            return userInput
+        # File cases
+        if os.path.isfile(userInput):
```

Comment:
Style nitpick: you could reduce the number of `if` levels but inverting some statements and returning early. For instance: `if not ... isfile: return None`.